backend/src/main/kotlin/com/infowings/catalog/data/objekt/ObjectService.kt

```diff
@@ -547,6 +564,23 @@ class ObjectService(
         deleteObject(id, username, ::removeOrMark)
     }
 
+    @Suppress("UNCHECKED_CAST")
+    fun recalculateValue(fromMeasureStr: String, toMeasureStr: String, value: BigDecimal): BigDecimal {
+        val fromMeasure = (GlobalMeasureMap[fromMeasureStr] ?: throw RecalculationException("Measure $fromMeasureStr does not exist")) as Measure<DecimalNumber>
+        val toMeasure = (GlobalMeasureMap[toMeasureStr] ?: throw RecalculationException("Measure $toMeasureStr does not exist")) as Measure<DecimalNumber>
+
+        val fromMeasureGroup =
+            MeasureMeasureGroupMap[fromMeasure.name] ?: throw IllegalStateException("Measure ${fromMeasure.name} does not belong to any measure group")
+        val toMeasureGroup =
+            MeasureMeasureGroupMap[toMeasure.name] ?: throw IllegalStateException("Measure ${toMeasure.name} does not belong to any measure group")
+
```

Comment:
Right hand sides of assignments to fromMeasureGroup/toMeasureGroup are pretty much the same.
It makes sense to define this logic as stand-alone method.

Reply:
In fact it is the same as to call `.getValue(key: String)` instead of flat `.get(key: String)`